overview/demo/shell.nix

```diff
@@ -32,7 +32,31 @@ let
         + ''
           export PS1="\[\033[1m\][demo-shell]\[\033[m\]\040\w >\040"
 
-          ${pkgs.lib.getExe pkgs.bash} --norc "$@"
+          ${lib.optionalString (demo-shell != { }) ''
+            echo -e "\n\033[1;32mDemo shell activated! Available programs:\033[0m"
+            ${lib.concatStringsSep "\n" (
+              lib.mapAttrsToList (
+                name: value:
+                lib.concatStringsSep "\n" (
+                  lib.mapAttrsToList (progName: prog: "echo '  - ${progName} (from ${name})'") value.programs
```

Comment:
I feel like it would be enough to print the executables without specifying which module they come from, unless we do something fancy like:

```
- xrsh
  - xrsh
- foobar
  - foobar-cli
  - foobard
```

But this can get a bit repetitive, nonetheless.